sway-lib-std/src/tx.sw

```diff
@@ -60,120 +114,116 @@ pub const OUTPUT_VARIABLE = 4u8;
 pub const OUTPUT_CONTRACT_CREATED = 5u8;
 
 /// Get the transaction type.
-pub fn tx_type() -> u8 {
-    asm(r1, r2: TX_TYPE_OFFSET) {
-        lw r1 r2 i0;
-        r1: u8
+pub fn tx_type(index: u64) -> u8 {
```

Comment:
`index` here is not really required right? From the spec, `index` is only useful and required for some of the tx fields like `GTF_SCRIPT_INPUT_AT_INDEX`

Reply:
yup, exactly.
>looks like I need to give this another pass to find all such fields and hardcode the index to 0 for them (an index is mandatory for gtf usage).